bin/compile

```diff
@@ -533,6 +538,77 @@ case "${TOOL}" in
             bin/go-post-compile
         fi
     ;;
+    goworkspaces)
```

Comment:
This looks like an exact copy of the gomodules section? I can't tell what's different. Is there a way to reduce the duplication here? Maybe goworkspaces and gomodules could share the same branch of the case statement?